core/src/workflows/hold_header.rs

```diff
@@ -0,0 +1,34 @@
+use crate::{
+    context::Context,
+    dht::actions::hold::hold_entry,
+    network::entry_with_header::EntryWithHeader,
+};
+
+use holochain_core_types::{
+    error::HolochainError,
+    chain_header::ChainHeader,
+    entry::Entry,
+};
+
+use holochain_persistence_api::cas::content::AddressableContent;
+
+use std::sync::Arc;
+
+pub async fn hold_header_workflow(
+    chain_header: &ChainHeader,
+    context: Arc<Context>,
+) -> Result<(), HolochainError> {
+
+    // 1. No need to validate. Store header in local DHT shard
+    // create an entry_with_header to leverage existing functionality. A headers header is itself.
+    let chain_header_entry = Entry::ChainHeader(chain_header.clone());
+    let entry_with_header = EntryWithHeader::new(chain_header_entry.clone(), chain_header.clone());
+    await!(hold_entry(&entry_with_header, context.clone()))?;
+
+    context.log(format!(
+        "debug/workflow/hold_header: HOLDING: {}",
+        chain_header_entry.address()
+    ));
+
+    Ok(())
+}
```

Comment:
Related to my comments above, I don't think we need a `hold_header` workflow for header publishing - instead headers should be first class entries that get stored through the regular `hold_entry` workflow.

There might a different future use-case for `hold_header` that works with the `EntryAspect::Header` that is actually storing *only* the header of an entry. This makes sense should we choose/need to implement something like *"sacriledging"* entries, i.e. making it a crime to speak about an entry, because the author has requested to have it deleted really because of privacy issues.. in that case we would still need to store the header and potentially other `EntryAspects` of that entry, to know that there was an entry with that address, but just not the content.